server/accounts_test.go

```diff
@@ -106,6 +107,195 @@ func TestAccountIsolation(t *testing.T) {
+		pubSubj  string
+	}{
+		{
+			name: "export literal, import literal",
+			exp:  "foo", imp: "foo",
+			pubSubj: "foo",
+		},
+		{
+			name: "export full wildcard, import literal",
+			exp:  "foo.>", imp: "foo.bar",
+			pubSubj: "foo.bar",
+		},
+		{
+			name: "export full wildcard, import sublevel full wildcard",
+			exp:  "foo.>", imp: "foo.bar.>",
+			pubSubj: "foo.bar.whizz",
+		},
+		// FAIL - dropping "foo.bar" from different account.
+		// What is the correct behavior here?
+		// Should this be allowed?
+		// If so, probably need to edit accounts.go:2427
+		// {
+		// 	name: "export literal, import full wildcard",
+		// 	exp:  "foo.bar", imp: "foo.>",
+		// 	pubSubj: "foo.bar",
+		// },
+		{
+			name: "export full wildcard, import full wildcard",
+			exp:  "foo.>", imp: "foo.>",
+			pubSubj: "foo.bar",
+		},
+		{
+			name: "export partial wildcard, import partial wildcard",
+			exp:  "foo.*", imp: "foo.*",
+			pubSubj: "foo.bar",
+		},
+		{
+			name: "export mid partial wildcard, import mid partial wildcard",
+			exp:  "foo.*.bar", imp: "foo.*.bar",
+			pubSubj: "foo.whizz.bar",
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			s := getServer(t)
+			defer s.Shutdown()
+			ncA, ncB := getAccountConns(t, s, c.exp, c.imp)
+			defer ncA.Close()
+			defer ncB.Close()
+
+			// ncA is the exporter account.
+			// ncB is the importer account.
+
+			// We keep track of 2 subjects.
+			// One subject (c.pubSubj) is based off the stream import.
+			// The other subject "fizz" is not imported and should be isolated.
+
+			gotSubjs := map[string]int{
+				c.pubSubj: 0,
+				"fizz":    0,
+			}
+			sub, err := ncB.Subscribe(">", func(m *nats.Msg) {
+				gotSubjs[m.Subject] += 1
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			time.Sleep(1 * time.Second)
+			if err := ncA.Publish(c.pubSubj, []byte(fmt.Sprintf("ncA pub %s", c.pubSubj))); err != nil {
+				t.Fatal(err)
+			}
+			if err := ncB.Publish(c.pubSubj, []byte(fmt.Sprintf("ncB pub %s", c.pubSubj))); err != nil {
+				t.Fatal(err)
+			}
+
+			if err := ncA.Publish("fizz", []byte("ncA pub fizz")); err != nil {
+				t.Fatal(err)
+			}
+			if err := ncB.Publish("fizz", []byte("ncB pub fizz")); err != nil {
+				t.Fatal(err)
+			}
+			time.Sleep(1 * time.Second)
+			if err := sub.Unsubscribe(); err != nil {
+				t.Fatal(err)
+			}
+
+			wantSubjs := map[string]int{
+				// Subscriber ncB should receive publishes from ncA and ncB.
+				c.pubSubj: 2,
+				// Subscriber ncB should only receive the publish from ncB.
+				"fizz": 1,
+			}
+			if got, want := len(gotSubjs), len(wantSubjs); got != want {
+				t.Fatalf("unexpected subjs len, got=%d; want=%d", got, want)
+			}
+
+			for key, gotCnt := range gotSubjs {
+				if wantCnt := wantSubjs[key]; gotCnt != wantCnt {
+					t.Errorf("unexpected receive count for subject %q, got=%d, want=%d", key, gotCnt, wantCnt)
+				}
+			}
+			if t.Failed() {
+				t.Logf("exported=%q; imported=%q", c.exp, c.imp)
+			}
```

Comment:
use t.Fatal and avoid t.Failed()

Reply:
Ah, so I'm using this differently than a `t.Fatal`. This if statement is to only print the export and import subjects if the previous loop encounters any errors.

You get an error log like this:
```
unexpected receive count for subject "what", got=1, want=2
unexpected receive count for subject "ever", got=3, want=4
unexpected receive count for subject "mysubj", got=5, want=6
exported="whatever"; imported="whatever"
```